fetch.bs

```diff
@@ -2172,6 +2264,216 @@ exploits pertain to those <a for=request>types</a>. Also, considering "<code>ima
+ using <var>providedTokenBindingId</var> and type <code>provided_token_binding</code>,
+ containing a signature (using <var>providedTokenBindingKeyPair</var>) over the
+ <var>providedTokenBindingId</var> as well as <var>tlsConnection</var>'s Exported Keying Material,
+ as described in
+ <a href="https://tools.ietf.org/html/draft-ietf-tokbind-protocol#section-3.3">section 3.3</a>
+ of [[!TOKBIND-PROTOCOL]].
+
+ <li><p>Set <var>tokenBindingMessage</var> to a list containing <var>providedTokenBinding</var>
+ as the only <a for=/>token binding</a>.
+
+ <li><p>If <var>httpRequest</var>'s <a for=request>use-referred-token-binding flag</a> is set,
+ then run these substeps:
+  <ol>
+   <li><p>Let <var>referredTlsConnection</var> be the result of getting a TLS connection for
+   <var>httpRequest</var>'s <a for=request>referred-token-binding origin</a>
+   with credentials <code>true</code> from the <a for=connection>connection pool</a>.
+
+   <li><p>If Token Binding was not negotiated for <var>referredTlsConnection</var>,
+   set <var>referredTokenBinding</var> to null.
+
+   <li><p>If Token Binding was negotiated for <var>referredTlsConnection</var> then:
+    <ol>
+    <li><p>Let <var>referredTokenBindingKeyPair</var> be the result of getting the Token Binding
+    key for <var>httpRequest</var>'s <a for=request>referred-token-binding origin</a>
+    and <var>referredTlsConnection</var>'s <a for=/>token-binding key parameters</a> metadata,
+    obtained by following the steps in the
+    <a href=#getting-token-binding-key>Getting a Token Binding Key</a> section of this document.
+
+    <li><p>Let <var>referredTokenBindingId</var> be the result of computing a
+    <a for=/>token binding ID</a> from <var>referredTokenBindingKeyPair</var>,
+    as described in
+    <a href="https://tools.ietf.org/html/draft-ietf-tokbind-protocol#section-3.2">section 3.2</a>
+    of [[!TOKBIND-PROTOCOL]].
+
+    <li><p>Let <var>referredTokenBinding</var> be the result of computing a
+    <a for=/>token binding</a> using <var>referredTokenBindingId</var> and type
+    <code>referred_token_binding</code>, containing a signature
+    (using <var>referredTokenBindingKeyPair</var>) over the <var>referredTokenBindingId</var>
+    as well as <var>tlsConnection</var>'s Exported Keying Material,
+    as described in
+    <a href="https://tools.ietf.org/html/draft-ietf-tokbind-protocol#section-3.3">section 3.3</a>
+    of [[!TOKBIND-PROTOCOL]].
+    </ol>
+
+   <li><p>If <var>referredTokenBinding</var> is not null, add it to
+   <var>tokenBindingMessage</var>.
+  </ol>
+
+  <li><p>If <var>tokenBindingMessage</var> is not null, compute and return its base64url-encoding
+  [[!RFC4648]]. Otherwise return null.
+</ol>
+Details of how to create a <a for=/>Token Binding Messsage</a> are described in
+<a href="https://tools.ietf.org/html/draft-ietf-tokbind-protocol#section-3">section 3</a>
+of [[!TOKBIND-PROTOCOL]].
```

Comment:
Perhaps you can merge this note with the final step above? That is, "... compute and return its base64url-encoding as described in section 3 of TOKBIND-PROTOCOL."